be/src/bench/shuffle_chunk_bench.cpp

```diff
@@ -0,0 +1,230 @@
+        int v = rand();
+        if (v % 100 < _null_percent) {
+            nullable_col->append_default();
+        } else {
+            nullable_col->append_datum(Datum("str123" + std::to_string(v)));
+        }
+    }
+    return c1;
+}
+
+ColumnPtr ShuffleChunkPerf::init_src_key_column(const TypeDescriptor& type) {
+    ColumnPtr c1 = ColumnHelper::create_column(type, true);
+    c1->reserve(_src_chunk_size);
+    auto* nullable_col = down_cast<NullableColumn*>(c1.get());
+    for (int k = 0; k < _src_chunk_size; k++) {
+        int v = rand();
+        nullable_col->append_datum(Datum("str123" + std::to_string(v)));
+    }
+    return c1;
+}
+
+ColumnPtr ShuffleChunkPerf::init_dest_column(const TypeDescriptor& type) {
+    auto c1 = ColumnHelper::create_column(type, true);
+    c1->reserve(_dest_chunk_size);
+    return c1;
+}
+
+ChunkPtr ShuffleChunkPerf::init_src_chunk() {
+    auto chunk = std::make_unique<Chunk>();
+    auto col = init_src_key_column(_types[0]);
+    chunk->append_column(col, 0);
+    for (int i = 1; i < _column_count; i++) {
+        col = init_src_column(_types[i]);
+        chunk->append_column(col, i);
+    }
+    return chunk;
+}
+
+void ShuffleChunkPerf::init_dest_chunks() {
+    for (int i = 0; i < _node_count; i++) {
+        if (_dest_chunks[i] == nullptr) {
+            _dest_chunks[i] = init_dest_chunk();
+        }
+    }
+}
+
+void ShuffleChunkPerf::reset_dest_chunks() {
+    for (int i = 0; i < _node_count; i++) {
+        if (_dest_chunks[i]->num_rows() > _dest_chunk_size) {
+            _dest_chunks[i].reset();
+        }
+    }
+}
+
+ChunkPtr ShuffleChunkPerf::init_dest_chunk() {
+    auto chunk = std::make_unique<Chunk>();
+    for (int i = 0; i < _column_count; i++) {
+        auto col = init_dest_column(_types[i]);
+        chunk->append_column(col, i);
+    }
+    return chunk;
+}
+
+void ShuffleChunkPerf::do_hash(const ColumnPtr& col) {
+    _shuffle_idxs.resize(_src_chunk_size);
+
+    col->crc32_hash(&_shuffle_idxs[0], 0, _src_chunk_size);
+    for (int i = 0; i < _src_chunk_size; i++) {
+        _shuffle_idxs[i] = _shuffle_idxs[i] % _node_count;
+    }
+}
+
+void ShuffleChunkPerf::do_shuffle(const Chunk& src_chunk) {
+    for (int n = 0; n < _node_count; n++) {
+        _select_idxs.resize(0);
+        for (int i = 0; i < _shuffle_idxs.size(); i++) {
+            if (_shuffle_idxs[i] == n) {
+                _select_idxs.emplace_back(i);
+            }
+        }
+        _dest_chunks[n]->append_selective(src_chunk, _select_idxs.data(), 0, _select_idxs.size());
+    }
+}
+
+void ShuffleChunkPerf::do_bench(benchmark::State& state) {
+    init_types();
+    _dest_chunks.resize(_node_count, nullptr);
+
+    state.ResumeTiming();
+
+    for (int i = 0; i < _chunk_count; i++) {
+        state.PauseTiming();
+        ChunkPtr src_chunk = init_src_chunk();
+        state.ResumeTiming();
+
+        do_hash(src_chunk->columns()[0]);
+
+        init_dest_chunks();
```

Comment:
If _chunk_count == 1, it will call state.ResumeTiming() twice?

Reply:
`init_src_chunk` is to generate test data and should not be included in performance tests。